android/src/main/java/com/google/samples/apps/ibeaconapp/lightbluebean/IBeaconManager.java

```diff
@@ -0,0 +1,73 @@
+package com.google.samples.apps.ibeaconapp.lightbluebean;
+
+import com.google.samples.apps.ibeaconapp.beaconinterface.IBeaconInterface;
+import nl.littlerobots.bean.Bean;
+import nl.littlerobots.bean.BeanDiscoveryListener;
+import nl.littlerobots.bean.BeanManager;
+
+import java.util.*;
+
+public class IBeaconManager implements IBeaconInterface {
+
+    private static volatile IBeaconManager instance;
+    private BeanDiscoveryListener beanDiscoveryListener;
+    private volatile HashMap<Bean, Integer> tempBeansAndRssi = new HashMap<Bean, Integer>();
+    private volatile HashMap<Bean, Integer> beansAndRssi = new HashMap<Bean, Integer>();
+    private long scanDelay = 4000L;
+    private final long timerDelay = 1000L;
+    private Timer mTimer;
+    private boolean isStarted = false;
+
+    public static IBeaconManager getInstance() {
+        IBeaconManager localInstance = instance;
+        if (localInstance == null) {
+            synchronized (IBeaconManager.class) {
+                localInstance = instance;
+                if (localInstance == null) {
+                    instance = localInstance = new IBeaconManager();
+                }
+            }
+        }
+        return localInstance;
+    }
+
+    public void startScanning() {
+        android.util.Log.w("test", "in start THREAD:" + Thread.currentThread().getName());
+        beanDiscoveryListener = new BeanDiscoveryListener() {
+            @Override
+            public void onBeanDiscovered(Bean bean, int rssi, List<UUID> list) {
```

Comment:
Use use nl.littlerobots.bean.Bean as a key in hashmap. I don't have nl.littlerobots.bean.Bean class sources locally and to lazy (and stupid) to download it, but does it for sure override hashCode() and equals() methods? Otherwise behaviour of code could be unexpected. Hashmaps, you know.